src/utils/time-utils.ts

```diff
@@ -152,37 +138,42 @@ const ifASingleDayChosen = (workDay: boolean, startWeek: number, endWeek: number
 /**
  * Get vacation days if multiple days are chosen
  *
- * @param workingWeek represents the working week
+ * @param workingWeek list of booleans representing which days are working days
  * @param vacationDayStart start week index of vacation
  * @param vacationDayEnd end week index of vacation
  * @param startWeek index of a start week day
  * @param endWeek index of the end week
  * @param weeks number of weeks
  */
-const ifADayRangeChosen = (workingWeek: boolean[], vacationDayStart: number, vacationDayEnd: number, startWeek: number, endWeek: number, weeks: number) => {
+const multipleVacationDaysSelected = (workingWeek: boolean[], vacationDayStart: number, vacationDayEnd: number, startWeek: number, endWeek: number, weeks: number) => {
   const workDays = workingWeek.filter(workDay => workDay).length;
-  if (vacationDayStart === vacationDayEnd){
-    if (endWeek - startWeek === 0) return weeks * 6 + 6;
-    if (!workingWeek[vacationDayEnd-1]) return weeks * 6;
+  const startsFromWorkingDay = workingWeek[vacationDayStart-1];
+
+  if (vacationDayStart === vacationDayEnd) {
+    if (!startsFromWorkingDay) return weeks * 6;
+    // vacation starts and ends on working day
     return weeks * 6 + 1;
   }
 
-  if (vacationDayEnd > vacationDayStart){
-    let addDay = 0;
+  if (vacationDayEnd > vacationDayStart) {
+    let takenWorkingDays = 0;
+    // calculate the number of vacation days from start till the end of vacation if the number of weeks is not an integer
     for (let i = vacationDayStart; i <= vacationDayEnd; i++){
-      if (workingWeek[i-1]) addDay+=1;
+      if (workingWeek[i-1]) takenWorkingDays+=1;
     }
-    if (addDay === workDays) addDay = 6;
-    return weeks * 6 + addDay;
+    if (takenWorkingDays === workDays) takenWorkingDays = 6;
+    return weeks * 6 + takenWorkingDays;
   }
 
-  let addDay = 0;
+  let takenWorkingDays = 0;
+  // calculate the number of vacation days from start vacation day till the end of working week if the number of weeks is not an integer
```

Comment:
Rather than "if the number of weeks is not an integer" could it be something like "if a full week is not taken". Talking about weeks not being an integer is a little confusing as the weeks variable in this function is of type number/ integer